implement WithUserOrder approches which provide the ability to export…

TaeZStkyoht opened the pull request:
… and mapping the yml file in user order.

and use c++ style casting,constness and declerations instead of c style.

Commit:
implement WithUserOrder approches which provide the ability to export and mapping the yml file in user order.
and use c++ style casting,constness and declerations instead of c style.

Commit:
Change libyaml version to 0.2.1 and MapObject

Commit:
Make switch's behaviour with the original one

jasonjei commented:
Thank you! I’ll review this weekend

Commit:
Add exportYamlWithUserOrderAndVersion, change wstring to const wstring&

Commit:
add _lineWithInfiniteWidth, update gitignore

Commit:
add yaml scalar style support, fix export user order start with sequence

Commit:
move yamlScalarStyle from mapPtr to MapObject

Commit:
prevent keys to be quoted

Commit:
add withversion functions

jasonjei commented:
I don't have any test cases written yet (lol) so I need to test this manually. I promise I'll take a look soon.

TaeZStkyoht commented:
Ok, thank you. I explained in commit messages what I did. I updated libyaml version, replaced c-style things, added new features without touching old version's behaviour, improved code. I developed in Linux. Of course, I also changed Windows defined parts, but it would be great to test it too. By the way, there are lots of code duplications. Maybe, I will fix that. But I need free time to do that. You can wait for those changes or of course you can do it too.

Commit:
generalize code, add CMake file, update gitignore

TaeZStkyoht commented:
@jasonjei One of my colleague made almost all of the last change. I also changed small part of it. You can now test and use. This is our final version of our contribution, at least for now.

TaeZStkyoht commented:
@jasonjei Did you look?
